services/galley/test/integration/API.hs

```diff
@@ -4416,31 +4416,7 @@ removeUser = do
         deleteUser alexDel' !!! const 200 === statusCode
 
     liftIO $ do
-      assertEqual ("expect exactly 7 federated requests in : " <> show fedRequests) 7 (length fedRequests)
-
-    liftIO $ do
-      bReq <- assertOne $ filter (matchFedRequest bDomain "on-user-deleted-conversations") fedRequests
-      frComponent bReq @?= Galley
-      frRPC bReq @?= "on-user-deleted-conversations"
-      Right udcnB <- pure . eitherDecode . frBody $ bReq
-      sort (fromRange (F.udcvConversations udcnB)) @?= sort [convB1, convB2]
-      F.udcvUser udcnB @?= qUnqualified alexDel
-
-    liftIO $ do
-      cReq <- assertOne $ filter (matchFedRequest cDomain "on-user-deleted-conversations") fedRequests
-      frComponent cReq @?= Galley
-      frRPC cReq @?= "on-user-deleted-conversations"
-      Right udcnC <- pure . eitherDecode . frBody $ cReq
-      sort (fromRange (F.udcvConversations udcnC)) @?= sort [convC1]
-      F.udcvUser udcnC @?= qUnqualified alexDel
-
-    liftIO $ do
-      dReq <- assertOne $ filter (matchFedRequest dDomain "on-user-deleted-conversations") fedRequests
-      frComponent dReq @?= Galley
-      frRPC dReq @?= "on-user-deleted-conversations"
-      Right udcnD <- pure . eitherDecode . frBody $ dReq
-      sort (fromRange (F.udcvConversations udcnD)) @?= sort [convD1]
-      F.udcvUser udcnD @?= qUnqualified alexDel
+      assertEqual ("expect exactly 4 federated requests in : " <> show fedRequests) 4 (length fedRequests)
```

Comment:
ok, the federated calls don't happen here any more, so we can't test for these events.  but should/can we test for something else?

related: a half-solution to the `MakesFederatedCall` tolerance for removing calls without removing the constraint could be that we add another kind of constraint `PushesFederatedNotifications`.  then we would get a type error if we forget to add the new constraint, and that would help us remember to remove the old one.  not sure that's worth it, though.

Reply:
> ok, the federated calls don't happen here any more, so we can't test for these events. but should/can we test for something else?

This functionality is already tested in federation end-to-end tests. If we _really_ want to do something we'd have to create a rabbitmq vhost (its like a namespace) and pass that to our test and then we can assert that there is something in the queue. But honestly, seems a bit redundant given there are other test. I am making sure that I don't delete tests without making sure that there are other tests.

Reply:
> related: a half-solution to the MakesFederatedCall tolerance for removing calls without removing the constraint could be that we add another kind of constraint PushesFederatedNotifications. then we would get a type error if we forget to add the new constraint, and that would help us remember to remove the old one. not sure that's worth it, though.

I thought the point behind adding `MakesFederatedCall` was to ensure that possibility of federation errors shows up in swagger. If we fail to connect to RabbitMQ, it is just a 500 because the backend is not well. Do you think it would help swagger somehow that we add this new annotation? Do the clients need to care that this gets send via RabbitMQ?